CONTRIBUTING.md

```diff
@@ -1,47 +1,297 @@
-# How to contribute
+# Contribution Guide #
+TiDB is a community driven open source project. All contributor are very welcome. The process of contributing to the TiDB project
+may be different than many other projects you get involved in. This document outlines some conventions about development workflow, commit message formatting, contact points and other resources to make it easier to get your contribution accepted. This document is the canonical source of truth for things like supported toolchain versions for building and testing TiDB.
 
-This document outlines some of the conventions on development workflow, commit message formatting, contact points and other
-resources to make it easier to get your contribution accepted.
+Please submit an [issue] on github if you 
+* find a requirement that this doc does not capture,
+* find other docs with references to requirements that
+  are not simply links to this doc.
 
-## Getting started
+## Pre submit flight checks
```

Comment:
It is a Metaphor. pre flight check is needed whenever you take a plane. Submit a PR is like flight. Pre checking is necessary to ensure PR in a good quality.